src/graph.h

```diff
@@ -343,4 +348,32 @@ struct DependencyScan {
   DyndepLoader dyndep_loader_;
 };
 
+// Prioritize edges by largest critical time first
```

Comment:
nit: This comment is confusing, because if this struct is used with std::sort(), it will actually place the edges with the smallest (not largest) critical time first, and otherwise smallest ids. A better comment would just state that instead.

The code works because std::priority_queue<> implements a maximal-heap, so pop() will always returns the "largest" item for the comparator being used, in this case, it will mean the one with the largest critical time, and otherwise the largest id. If all critical times are equal, e.g. for a clean build, this reverts the order currently used by the implementation, this is subtle, and may be even undesirable (does it make clean builds slower?).